pkg/landscaper/blueprints/resolve.go

```diff
@@ -124,18 +131,25 @@ func Resolve(ctx context.Context, registryAccess model.RegistryAccess, cdRef *ls
 	if err != nil {
 		return nil, fmt.Errorf("unable to resolve component descriptor for ref %#v: %w", cdRef, err)
 	}
-
-	return ResolveBlueprintFromComponentVersion(ctx, componentVersion, bpDef.Reference.ResourceName)
+	resource, err := componentVersion.GetResource(bpDef.Reference.ResourceName, nil)
+	if err != nil {
+		return nil, err
+	}
+	content, err := resource.GetTypedContent(ctx)
+	if err != nil {
+		return nil, err
+	}
+	return content.Resource.(*Blueprint), nil
```

Comment:
A cast without a check if critical as it will cause a panic stopping the whole system.

Reply:
I think we can be certain that we have a blueprint at this point. But you are right, better safe than sorry! Added checks to each occurrence.